custom_components/localtuya/__init__.py

```diff
@@ -478,6 +478,10 @@ async def device_state_changed(event: Event[dr.EventDeviceRegistryUpdatedData]):
 
         dev_id = _device_id_by_identifiers(device_registry.identifiers)
         host_ip = entry.data[CONF_DEVICES][dev_id][CONF_HOST]
+
+        if cid := entry.data[CONF_DEVICES][dev_id].get(CONF_NODE_ID):
```

Comment:
Why do you use different notations to get `host_ip` (`[CONF_HOST]`) and `cid` (`get(CONF_NODE_ID)`) in subsequent lines?

Reply:
Because HOST is always exists unlike CID where is possible that it's missing and try to access would raise an error.